kernel/examples/read-table-single-threaded/src/main.rs

```diff
@@ -114,7 +114,8 @@ fn try_main() -> DeltaResult<()> {
         .build()?;
 
     let mut batches = vec![];
-    for res in scan.execute(engine.as_ref())?.into_iter() {
+    for res in scan.execute(engine.as_ref())? {
+        let res = res?;
         let data = res.raw_data?;
```

Comment:
For this and similar changes, I was looking to avoid multiple calls to `?` or `.unwrap()` on the same value, since res is used a couple times. Felt cleaner that way. So we're fine with multiple calls?

Reply:
right, i realized that later. i think then you can just do `let res = res?` and you only have to do it once.